library/coverage/result_spec.rb

```diff
@@ -1,33 +1,95 @@
 require File.expand_path('../../../spec_helper', __FILE__)
 require 'coverage'
 
+# Clear old results from the result hash
+# https://bugs.ruby-lang.org/issues/12220
+def filtered_result
+  Coverage.result.select { |_k, v| v.any? }
+end
+
 describe 'Coverage.result' do
   before :all do
+    @class_file = fixture __FILE__, 'some_class.rb'
     @config_file = fixture __FILE__, 'start_coverage.rb'
   end
 
   after :each do
+    $LOADED_FEATURES.delete(@class_file)
     $LOADED_FEATURES.delete(@config_file)
   end
 
-  it 'gives the covered files as a hash with arrays' do
+  it 'gives the covered files as a hash with arrays of count or nil' do
+    Coverage.start
+    require @class_file.chomp('.rb')
+    result = filtered_result
+
+    result.should == {
+      @class_file => [
+        nil, nil, 1, nil, nil, 1, nil, nil, 0, nil, nil, nil, nil, nil, nil, nil
+      ]
+    }
+  end
+
+  it 'no requires/loads should give empty hash' do
+    Coverage.start
+    result = filtered_result
+
+    result.should == {}
+  end
+
+  it 'second call should give exception' do
+    Coverage.start
+    require @class_file.chomp('.rb')
+    Coverage.result
+    begin
+      Coverage.result
```

Comment:
Could not find an example that also verifies the message.  Just expecting RuntimeError is not enough, is it?  The message is the important part since RuntimeError is very generic.